.github/workflows/deploy-changed-cf.yaml

```diff
@@ -16,49 +26,87 @@ env:
+          fi
+
+      - id: set-matrix
+        name: Set up cluster matrix
+        run: |-
+          # Github workflow_dispatch inputs do not use use default value if workflow is not
+          # triggered by dispatch. Therefore, we have to manually provide a default value
+          CLUSTER=${CLUSTER:-"staging"}
+
+          if [ "${GITHUB_REF}" == "refs/heads/master" ] || [ "${CLUSTER}" = "staging and production" ]; then
+            echo '::set-output name=matrix::["staging", "production"]'
+          elif [ "${CLUSTER}" = "staging" ]; then
+            echo '::set-output name=matrix::["staging"]'
+          elif [ "${CLUSTER}" = "production" ]; then
+            echo '::set-output name=matrix::["production"]'
+          fi
+        env:
+          CLUSTER: ${{ github.event.inputs.cluster }}
+          GITHUB_REF: ${{ github.ref }}
+
   get-modified-templates:
     name: Fetch paths to modified CloudFormation templates
     runs-on: ubuntu-20.04
     outputs:
       files: ${{ steps.check-number-of-cf-files.outputs.files }}
       num-files: ${{ steps.check-number-of-cf-files.outputs.num-files }}
     steps:
+      - id: checkout
+        name: Check out current branch source code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
       - id: get-changed-files
         name: Get changed files
-        uses: lots0logs/gh-action-get-changed-files@2.1.4
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          echo "[]" > empty.json
 
-      - id: check-number-of-cf-files
-        name: Check CloudFormation templates
-        run: |-
-          # Merge all modified and renamed files into one JSON array
-          #jq -s add ${HOME}/files_added.json ${HOME}/files_modified.json ${HOME}/files_renamed.json > ${HOME}/files_to_check.json
+          # Get name of changed files
+          # and concatenate into a JSON array
+          git diff --name-only --diff-filter=ACMRT remotes/origin/master ${{ github.sha }} | xargs -I 'value' jq --arg filename "value" '. |= . + [$filename]' empty.json | jq -s 'add' > ${HOME}/files.json
 
+          echo "All changed files"
           cat ${HOME}/files.json
-          echo
 
+      - id: check-number-of-cf-files
+        name: Check CloudFormation templates
+        run: |-
           # Select those that are CF templates (path starts with `cf/`)
           jq '[.[] | select(match("^cf/"))]' ${HOME}/files.json > ${HOME}/cf_files.json
 
+          echo "Changed CF files"
+          cat ${HOME}/cf_files.json
+
           # Set as output the minified JSON.
           echo "::set-output name=files::$(jq -c . < ${HOME}/cf_files.json)"
           echo "$(jq -c . < ${HOME}/cf_files.json)"
           echo "::set-output name=num-files::$(jq '. | length' ${HOME}/cf_files.json)"
 
-  deploy-templates:
-    name: Deploy changed CloudFormation template
+  lint-templates:
+    name: Lint template files
```

Comment:
why did you change this?

Reply:
I split this job to be independent from the deploy job.